genesis/src/lib.rs

```diff
@@ -23,9 +26,15 @@ pub static GENESIS_DOMAIN_ID: Lazy<DomainId> = Lazy::new(|| "genesis".parse().un
 #[repr(transparent)]
 pub struct GenesisTransaction(pub SignedTransaction);
 
+/// Genesis block.
+/// First instruction should be [`Upgrade`].
```

Comment:
Is this enforced? Also I believe genesis can only have 1 transaction?

Reply:
It is checked in `ValidBlock::validate` that genesis block has single transaction. Added check that first instruction is `Upgrade`